BetterR1.txt

```diff
@@ -40,34 +39,30 @@ class Filter:
             modified_messages.append(modified_message)
 
         body["messages"] = modified_messages
-
-        # print(f"inlet:modified_body:{body}")
-
         return body
 
     async def outlet(
         self, body: dict, __user__: Optional[dict] = None, __event_emitter__=None
     ) -> dict:
-        # print(f"outlet:{__name__}")
-        # print(f"outlet:body:{body}")
-        # print(f"outlet:user:{__user__}")
-
+        elapsed_time = time.time() - Filter.start_time
+        
         messages = body.get("messages", [])
-
         if not messages:
             return body
 
-        User_message = messages[-2].get("content", "")
-
         model_response = messages[-1].get("content", "")
+        thought_time_str = f"Thought for {int(elapsed_time)} seconds...\n\n"
+        model_response = thought_time_str + model_response
 
         model_response = model_response.replace(
             "<think>", "<details>\n<summary>Thoughts 💭</summary>"
         )
-
         model_response = model_response.replace("</think>", "\n\n---\n\n</details>")
 
         messages[-1]["content"] = model_response
         body["messages"] = messages
 
+        # Reset timer for next run
+        Filter.start_time = time.time()
```

Comment:
This is being carried over to the next run... maybe set it to none and start on the inlet if none.